ddl/reorg.go

```diff
@@ -206,25 +196,24 @@ func (w *worker) runReorgJob(t *meta.Meta, reorgInfo *reorgInfo, tblInfo *model.
 			Location:      &model.TimeZoneLocation{Name: time.UTC.String(), Offset: 0},
 		}
 	}
-	if w.reorgCtx.doneCh == nil {
+	// this job is cancelling, we should notify the reorg worker.
+	if job.IsCancelling() {
```

Comment:
If `job.IsCancelling()` is true, `rc.isReorgCanceled()` is true too. So we will do `removeReorgCtx` at this [line](https://github.com/pingcap/tidb/pull/33878/files#diff-51926b25b437247805c6880b2fd041f096575b30050b560ab934f1d863bb6bf6R235). 
Then could we remove this logic?